### DISCUSSION TITLE
Load config file which is located outside package directory

### DISCUSSION THREAD

**Discussion by wigging** [pallets/flask#5429] [Q&A] [answered]
I have a simple flask project as shown below. There is a config file `config.json` where things like port, host, debug, etc. can be defined.

```
flask-example
├── README.md
├── config.json
├── pyproject.toml
└── src/
    └── flaskr/
        ├── __init__.py
```

In the `__init__.py` I have the following:

```python
import json
from flask import Flask

def create_app():
    app = Flask(__name__)

    app.config.from_file("config.json", load=json.load)

    @app.route("/hello")
    def hello():
        return "Hello, World!"

    return app
```

The configuration file `config.json` is located outside the package source code directory. The content of the config file is:

```json
{
    "PORT": 5040,
    "DEBUG": true
}
```

When I run ` flask --app flaskr run`, I get a file not found error because flask is looking in the `src/flaskr` directory for the config file. How do I provide the path to the config file so I don't have to put the file inside the package source?

**Comment by tachyondecay** [comment, marked as answer]
The simplest answer is: use an absolute path.

Relative paths are relative to the app config object's root path, which by default will be `app.root_path`, which can be a couple of different things depending on how your package is configured. If you’re using a `src` layout, you might want to install your package in editable mode.

**Reply by wigging** [reply]
Using the absolute path to the config file does not work. Flask is able to load the file, but when I run the app it still uses the default 5000 port and debug mode is off.

**Reply by davidism** [reply]
That's because those values are set in the server, they're not part of app config.

**Reply by wigging** [reply]
So how would I set these values using a configuration file?

**Reply by davidism** [reply]
You wouldn't. You set them on the command line when running the dev server.

**Reply by wigging** [reply]
So I should use the config file for app settings and configure the server from the command line?

**Reply by ThiefMaster** [reply]
Yes. The dev server is ONLY for development anyway. You never use `flask run` or `app.run()` outside development.
